pulp_ansible/app/tasks/collections.py

```diff
@@ -674,21 +674,46 @@ async def _add_namespace(self, name):
             pass
         else:
             namespace = parse_metadata(result)
-            namespace.pop("pulp_href", None)
-            url = namespace.pop("avatar_url", None)
-            da = (
-                [
-                    DeclarativeArtifact(
-                        Artifact(sha256=namespace["avatar_sha256"]),
-                        url=url,
-                        remote=self.remote,
-                        relative_path=f"{name}-avatar",
-                        deferred_download=False,
-                    )
-                ]
-                if url
-                else None
-            )
+            links = namespace.get("links", None)
+
+            # clean up the galaxy API for pulp
```

Comment:
The special code for handling `avatar_url` and `avatar_sha256` doesn't really make sense to me. This method `_add_namespace` only gets called if the collection version metadata has `metadata_sha256` in it's `namespace` field (see line 658), so syncing against current galaxy_ng wouldn't trigger this. If you want namespace sync to work against a remote that doesn't have the pulp_ansible NamespaceMetadata code, then you are going to have to add some more hacks.

Reply:
I added the `metadata_sha256` to the namespace API in galaxy (https://github.com/ansible/galaxy_ng/pull/1642/files#diff-7f24d0266c5878d82a98acb7d91612b64915a8209671edd13c57478283d119a0R53).

 Unfortunately we DO get `metadata_sha256` on the V3 collection apis because I migrated the existing ns metadata as part of https://github.com/pulp/pulp_ansible/pull/1395. I could remove the migration, but we'll only get NS metadata on the search api if there's a metadata object for it.

I thought about trying to provide the avatar_sha on galaxy_ng, but it's too complicated. I'd have to figure out how to download all of the images that are currently only provided as URLs. This approach seemed a lot more feasible, and as a bonus we get logo sync.